airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultNormalizationWorker.java

```diff
@@ -58,10 +66,16 @@ public NormalizationSummary run(final NormalizationInput input, final Path jobRo
 
       if (!normalizationRunner.normalize(jobId, attempt, normalizationRoot, input.getDestinationConfiguration(), input.getCatalog(),
           input.getResourceRequirements())) {
-        throw new WorkerException("Normalization Failed.");
+        normalizationRunner.getTraceMessages()
+            .filter(traceMessage -> traceMessage.getType() == AirbyteTraceMessage.Type.ERROR)
+            .forEach(traceMessage -> traceFailureReasons.add(FailureHelper.normalizationFailure(traceMessage, Long.valueOf(jobId), attempt)));
+        failed = true;
       }
     } catch (final Exception e) {
-      throw new WorkerException("Normalization Failed.", e);
+      normalizationRunner.getTraceMessages()
```

Comment:
Can we extract this block and the block from line 69 to 72 to a method?

Reply:
👍